openmc/mgxs/mgxs.py

```diff
@@ -4809,64 +4849,953 @@ class Chi(MGXS):
+    scores : list of str
+        The scores in each tally used to compute the multi-group cross section
+    filters : list of openmc.Filter
+        The filters in each tally used to compute the multi-group cross section
+    tally_keys : list of str
+        The keys into the tallies dictionary for each tally used to compute
+        the multi-group cross section
+    estimator : 'analog'
+        The tally estimator used to compute the multi-group cross section
+    tallies : collections.OrderedDict
+        OpenMC tallies needed to compute the multi-group cross section. The keys
+        are strings listed in the :attr:`ConsistentScatterMatrixXS.tally_keys`
+        property
+        and values are instances of :class:`openmc.Tally`.
+    rxn_rate_tally : openmc.Tally
+        Derived tally for the reaction rate tally used in the numerator to
+        compute the multi-group cross section. This attribute is None
+        unless the multi-group cross section has been computed.
+    xs_tally : openmc.Tally
+        Derived tally for the multi-group cross section. This attribute
+        is None unless the multi-group cross section has been computed.
+    num_subdomains : int
+        The number of subdomains is unity for 'material', 'cell' and 'universe'
+        domain types. This is equal to the number of cell instances
+        for 'distribcell' domain types (it is equal to unity prior to loading
+        tally data from a statepoint file).
+    num_nuclides : int
+        The number of nuclides for which the multi-group cross section is
+        being tracked. This is unity if the by_nuclide attribute is False.
+    nuclides : Iterable of str or 'sum'
+        The optional user-specified nuclides for which to compute cross
+        sections (e.g., 'U238', 'O16'). If by_nuclide is True but nuclides
+        are not specified by the user, all nuclides in the spatial domain
+        are included. This attribute is 'sum' if by_nuclide is false.
+    sparse : bool
+        Whether or not the MGXS' tallies use SciPy's LIL sparse matrix format
+        for compressed data storage
+    loaded_sp : bool
+        Whether or not a statepoint file has been loaded with tally data
+    derived : bool
+        Whether or not the MGXS is merged from one or more other MGXS
+    hdf5_key : str
+        The key used to index multi-group cross sections in an HDF5 data store
+
+    """
+
+    def __init__(self, domain=None, domain_type=None, groups=None,
+                 by_nuclide=False, name='', num_polar=1,
+                 num_azimuthal=1, nu=False):
+        super(ConsistentScatterMatrixXS, self).__init__(
+            domain, domain_type, groups, by_nuclide=by_nuclide,
+            name=name, num_polar=num_polar, num_azimuthal=num_azimuthal)
+
+        self.nu = nu
+
+        # Initialize each MGXS used by the convolution
+        self._mgxs = \
+            [ScatterXS(), ScatterProbabilityMatrix(), MultiplicityMatrixXS()]
```

Comment:
It looks like this scheme requires four different analog tallies (`nu-scatter-0` and `scatter-0` with energy in/out filters for multiplicity, and `nu-scatter-pN` [or with histogram bins] with energy/inout filters for the scatter probability matrix, `flux` w/ energyin filter) and two tracklength tallies (`flux` and `scatter-0` both with an energyin filter).  

If you coded this to directly deal with the tally data instead of going through the pre-existing MGXS classes then you could drop the analog `nu-scatter-0` with energy in/out filters, and the analog `flux`.  
I think this could be done with:
`psuedo_tl_nu-scatter[gin, gout, n] = tl_scatter-0[gin] / tl_flux[gin] *  (analog_nu-scatter-pN[gin, gout, 0] / analog_scatter-0[gin, gout]) * (analog_nu-scatter-pN[gin, gout, n] / sum_{gout=1:G}(analog_nu-scatter-pN[gin, gout, 0]))`
(of course the above only works for Legendre scattering, not histogram, but thats only a minor tweak)

That is a big deal for OpenMC runtime performance.  In addition, doing so obviates the need for the generality of the Convolved class, meaning I think this can just be an extended type of `ScatterMatrixXS`, which I'm also obviously a fan of! ;-)